cloud/services/virtualnetworks/service.go

```diff
@@ -18,14 +18,15 @@ package virtualnetworks
 
 import (
 	"github.com/go-logr/logr"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
 	azure "sigs.k8s.io/cluster-api-provider-azure/cloud"
 )
 
 // VNetScope defines the scope interface for a virtual network service.
 type VNetScope interface {
 	logr.Logger
 	azure.ClusterDescriber
-	azure.NetworkDescriber
+	Vnet() *infrav1.VnetSpec
```

Comment:
I think this highlights a potential bug in the code... the vnet service is supposed to be unaware of the AzureCluster spec and just create / update / delete vnets from the specs it gets. The fact that it needs `Vnet()` which doesn't have any info that `VNetSpecs()` doesn't have is a little strange. Looking at the code, it uses that to write the info back into the Vnet() which makes sense but also would cause a bug if we have decided to have multiple specs returned by VnetSpecs() since they would overwrite each other. Not specific to this PR, in fact this PR makes it easy to see that bug, just reflecting 🤔

Reply:
this is another case where it would make more sense if vnetSpec() didn't return an array @alexeldeib

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1001#issuecomment-718980544